<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y> 
    <width>817</width>
    <height>712</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QCodeEdit</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QStackedWidget" name="m_stack">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="m_welcomePage">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="m_logo">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;



&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;



p, li { white-space: pre-wrap; }



&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;



&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Welcome to &lt;span style=&quot; font-weight:600;&quot;&gt;QCodeEdit 2&lt;/span&gt; demo application.&lt;/p&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;



&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;It is a lightweight but fully functional source code editor which support a wide range of languages and make it easy to add support for other languages through XML-based syntax files.&lt;/p&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;



&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Hoping you'll like it.&lt;/p&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="indent">
           <number>-1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCommandLinkButton" name="clbEdit">
          <property name="text">
           <string>Ok let's roll!</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QStackedWidget" name="welcomeStack">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QCommandLinkButton" name="clbMore">
              <property name="text">
               <string>Learn more</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>38</width>
                <height>339</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QCommandLinkButton" name="clbLess">
              <property name="text">
               <string>Less</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;QCodeEdit is a library based on Nokia's (formerly Trolltech's) Qt 4 toolkit. It is available for free under GNU General Public License version 3. It is also possible to buy licenses compatible with proprietary software (e.g BSD).&lt;/p&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;



&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Thanks to all the testers and particularly to those who contributed patches or made suggestion. The most helpful ones so far, in no particular order : &lt;b&gt;Benito van der Zander&lt;/b&gt; (TexMakerX : LaTex editor), &lt;b&gt;Ulrich van den Hekke&lt;/b&gt; (Xinx : XSL/JavScript editor), &lt;b&gt;Boris Barbulovski&lt;/b&gt; (ColdFusion editor), &lt;b&gt;Phil Martinot&lt;/b&gt; (Lua editor?), &lt;b&gt;Jeremy Sonander&lt;/b&gt; (proprietary VHDL editor), &lt;b&gt;Jerome Vizcaino&lt;/b&gt; (assembly editor)&lt;/p&gt;



&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;/p&gt;



&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Funders also deserve big thanks because QCodeEdit wouldn't have gone that far without them (money can buy time sometimes) :&lt;/p&gt;&lt;ul&gt;&lt;li&gt;Saros Inc &lt;/li&gt;&lt;li&gt;Movimento&lt;/li&gt;&lt;li&gt;Averant (through Saros)&lt;/li&gt;&lt;/ul&gt;



</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>



&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;QCodeEdit is Copyright Â© 2007-2009 Luc BRUANT aka fullmetalcoder&lt;/p&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Maximum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_settingsPage">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QListWidget" name="lwSettings">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>128</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>128</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="autoScrollMargin">
           <number>0</number>
          </property>
          <property name="showDropIndicator" stdset="0">
           <bool>false</bool>
          </property>
          <property name="alternatingRowColors">
           <bool>false</bool>
          </property>
          <property name="iconSize">
           <size>
            <width>96</width>
            <height>96</height>
           </size>
          </property>
          <property name="textElideMode">
           <enum>Qt::ElideNone</enum>
          </property>
          <property name="movement">
           <enum>QListView::Static</enum>
          </property>
          <property name="flow">
           <enum>QListView::LeftToRight</enum>
          </property>
          <property name="isWrapping" stdset="0">
           <bool>true</bool>
          </property>
          <property name="resizeMode">
           <enum>QListView::Adjust</enum>
          </property>
          <property name="spacing">
           <number>10</number>
          </property>
          <property name="viewMode">
           <enum>QListView::IconMode</enum>
          </property>
          <property name="uniformItemSizes">
           <bool>false</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="selectionRectVisible">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>General</string>
           </property>
           <property name="icon">
            <iconset resource="example.qrc">
             <normaloff>:/settings.png</normaloff>:/settings.png</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Editor</string>
           </property>
           <property name="icon">
            <iconset resource="example.qrc">
             <normaloff>:/editor.png</normaloff>:/editor.png</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Syntax</string>
           </property>
           <property name="icon">
            <iconset resource="example.qrc">
             <normaloff>:/formats.png</normaloff>:/formats.png</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string/>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QStackedWidget" name="settingsStack">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <widget class="QWidget" name="page_4">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>Display</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <item row="2" column="0">
                <widget class="QCheckBox" name="chkLineNumbers">
                 <property name="text">
                  <string>Show line numbers</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QCheckBox" name="chkFoldPanel">
                 <property name="text">
                  <string>Show fold indicators</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QCheckBox" name="chkLineChange">
                 <property name="text">
                  <string>Show line change panel</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QCheckBox" name="chkLineMarks">
                 <property name="text">
                  <string>Show line marks panel</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QCheckBox" name="chkStatusPanel">
                 <property name="text">
                  <string>Show status panel</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="5" column="0" colspan="2">
                <spacer name="verticalSpacer_5">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeType">
                  <enum>QSizePolicy::Fixed</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="6" column="0">
                <widget class="QCheckBox" name="chkWrap">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>Dynamic line wrapping</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="1">
                <widget class="QCheckBox" name="chkMoveInWrap">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Cursor movement within wrapped blocks</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="7" column="0">
                <widget class="QCheckBox" name="chkIndent">
                 <property name="text">
                  <string>Auto indent</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_2">
              <property name="title">
               <string>Files</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_3">
               <item row="0" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Recent files</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spnRecent">
                 <property name="maximum">
                  <number>25</number>
                 </property>
                 <property name="value">
                  <number>10</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_4">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>377</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item row="0" column="3">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>528</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="4">
         <widget class="QDialogButtonBox" name="bbSettings">
          <property name="standardButtons">
           <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::RestoreDefaults</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_snippetsPage">
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QSnippetEdit" name="m_snippetEdit" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>817</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menu_Recent">
     <property name="title">
      <string>&amp;Recent</string>
     </property>
     <addaction name="action_Clear"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="menu_Recent"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="action_SaveAs"/>
    <addaction name="separator"/>
    <addaction name="action_Print"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_Advanced">
    <property name="title">
     <string>&amp;Advanced</string>
    </property>
    <addaction name="action_Settings"/>
    <addaction name="action_Snippets"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
    <addaction name="action_About_Qt"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Advanced"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="mainToolbar">
   <property name="windowTitle">
    <string>File</string>
   </property>
   <property name="iconSize">
    <size>
     <width>24</width>
     <height>24</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="separator"/>
   <addaction name="action_Save"/>
   <addaction name="action_SaveAs"/>
   <addaction name="separator"/>
   <addaction name="action_Print"/>
   <addaction name="separator"/>
   <addaction name="action_Editor"/>
   <addaction name="action_Settings"/>
   <addaction name="action_Snippets"/>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/open.png</normaloff>:/open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/exit.png</normaloff>:/exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_About_Qt">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
  <action name="action_Settings">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/settings.png</normaloff>:/settings.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Settings</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_Reload_syntax_files">
   <property name="text">
    <string>&amp;Reload syntax files</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/save.png</normaloff>:/save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_SaveAs">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/saveas.png</normaloff>:/saveas.png</iconset>
   </property>
   <property name="text">
    <string>Save &amp;as</string>
   </property>
  </action>
  <action name="action_Print">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/print.png</normaloff>:/print.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Print</string>
   </property>
   <property name="toolTip">
    <string>Print</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_Clear">
   <property name="text">
    <string>&amp;Clear</string>
   </property>
  </action>
  <action name="action_New">
   <property name="icon">
    <iconset resource="example.qrc">
     <normaloff>:/new.png</normaloff>:/new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_Snippets">
   <property name="text">
    <string>Snippets</string>
   </property>
  </action>
  <action name="action_Editor">
   <property name="text">
    <string>Editor</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="2" margin="2"/>
 <customwidgets>
  <customwidget>
   <class>QSnippetEdit</class>
   <extends>QWidget</extends>
   <header>qsnippetedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>clbEdit</tabstop>
  <tabstop>clbMore</tabstop>
  <tabstop>lwSettings</tabstop>
  <tabstop>chkLineNumbers</tabstop>
  <tabstop>chkFoldPanel</tabstop>
  <tabstop>chkLineChange</tabstop>
  <tabstop>chkLineMarks</tabstop>
  <tabstop>chkStatusPanel</tabstop>
  <tabstop>chkWrap</tabstop>
  <tabstop>chkMoveInWrap</tabstop>
  <tabstop>chkIndent</tabstop>
  <tabstop>spnRecent</tabstop>
  <tabstop>bbSettings</tabstop>
  <tabstop>clbLess</tabstop>
 </tabstops>
 <resources>
  <include location="example.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lwSettings</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>settingsStack</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkWrap</sender>
   <signal>toggled(bool)</signal>
   <receiver>chkMoveInWrap</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>226</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
